skrub/_join_aggregator.py

```diff
@@ -0,0 +1,354 @@
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+NUM_OPS = ["mean", "std", "max", "min"]
+CATEG_OPS = ["mode"]
+ALL_OPS = NUM_OPS + CATEG_OPS
+
+
+def split_num_categ_ops(agg_ops):
+    """Separate aggregagor operators input 
+    by their type.
+
+    Parameters
+    ----------
+    agg_ops : list of str,
+        The input operators names.
+
+    Returns
+    -------
+    num_ops, categ_ops : Tuple of List of str
+        List of operator names
+    """
+    num_ops, categ_ops = [], []
+    for op_name in agg_ops:
+        if op_name in NUM_OPS:
+            num_ops.append(op_name)
+        elif op_name in CATEG_OPS:
+            categ_ops.append(op_name)
+        else:
+            ValueError(
+                f"'ops' options are {ALL_OPS}, got: {op_name}."
+            )
+    return num_ops, categ_ops
+
+
+class AssemblingEngine:
+    """Helper class to perform the join and aggregate operations.
+
+    This is an abstract base class that is specialized depending
+    on the module of the dataframe used (Pandas or Polars).
+
+    This helper class is in charge of mapping strings to aggregation
+    operators of the respective module.
+    """
+    @classmethod
+    def get_for(cls, tables):
+        """Returns the AssemblingEngine implementation for given tables
+        module.
+
+        Parameters
+        ----------
+        tables : List of Tuple of (table, cols_to_join, cols_to_agg)
+            We only use table to detect the module used.
+        
+        Returns
+        -------
+        assembling_engine : AssemblingEngine
+            The suited AssemblingEngine implementation.
+        """
+        use_pandas = all(
+            [isinstance(table, pd.DataFrame) for table, _, _ in tables]
+        )
+        use_polars = False
+        if POLARS_SETUP:
+            # we don't mix DataFrame and LazyFrame
+            use_polars = (
+                all(
+                    [isinstance(table, pl.DataFrame) for table, _, _ in tables]
+                )
+                or all(
+                    [isinstance(table, pl.LazyFrame) for table, _, _ in tables]
+                )
+            )
+        if use_pandas:
+            return PandasAssemblingEngine()
+        elif use_polars:
+            return PolarsAssemblingEngine()
+        else:
+            raise NotImplementedError(
+                "Only Pandas or Polars DataFrame are currently supported."
+            )
+
+    @abstractmethod
+    def agg(self, table, cols_to_join, cols_to_agg, ops):
+        pass
+
+    @abstractmethod
+    def join(self, left, right, left_on, right_on):
+        pass
+
+
+def pandas_split_num_categ_cols(table):
+    num_cols = table.select_dtypes("number").columns
+    categ_cols = table.select_dtypes(
+        ["object", "string", "category"]
+    ).columns
+    return num_cols, categ_cols
+    
+
+def pandas_get_agg_ops(cols, agg_ops):
+    pandas_ops_mapping = {
```

Comment:
To stay consistent, we'd like to also inline the `{pandas, polars}_get_agg_ops` and `{pandas, polars}_split_num_categ_cols` functions. 

However, some of these functions are long and might clutter the logic of the calling method. 

Maybe we should define these functions as `staticmethod`? So that things are close to each other.

WDYT?